src/note/nullifier.rs

```diff
@@ -12,6 +14,22 @@ use crate::{
 pub struct Nullifier(pub(super) pallas::Base);
 
 impl Nullifier {
+    /// Generates a dummy nullifier for use as $\rho$ in dummy spent notes.
+    ///
+    /// Nullifiers are required by consensus to be unique. For dummy output notes, we get
+    /// this restriction as intended: the note's $\rho$ value is set to the nullifier of
+    /// the accompanying spent note within the action, which is constrained by consensus
+    /// to be unique. In the case of dummy spent notes, we get this restriction by
+    /// following the chain backwards: the nullifier of the dummy spent note will be
+    /// constrained by consensus to be unique, and the nullifier's uniqueness is derived
+    /// from the uniqueness of $\rho$.
+    ///
+    /// Instead of explicitly sampling for a unique nullifier, we rely here on the size of
+    /// the base field to make the chance of sapling a colliding nullifier negligible.
+    pub(crate) fn dummy(rng: &mut impl RngCore) -> Self {
+        Nullifier(pallas::Base::random(rng))
```

Comment:
```suggestion
        Nullifier(extract_p(&pallas::Point::random(rng)))
```
(needs an import of `group::Group`).